chainer/testing/unary_function_test.py

```diff
@@ -0,0 +1,179 @@
+       value domain leaving some margin around zero of input ``x``.
+
+       >>> def make_data(dtype, shape):
+       >>>     x = numpy.random.uniform(0.1, 1, shape).astype(dtype)
+       >>>     gy = numpy.random.uniform(-1, 1, shape).astype(dtype)
+       >>>     return x, gy
+       >>>
+       >>> @unary_function_test(F.sqrt, make_data=make_data)
+       >>> class TestSqrt(unittest.TestCase):
+       >>>     pass
+
+       We define ``make_data`` function to return input and gradient ndarrays
+       generated in proper value domains with given ``dtype`` and ``shape``
+       parameters, then passing it to the decorator's ``make_data`` argument.
+
+    """
+
+    # Import here to avoid mutual import.
+    from chainer import gradient_check
+    from chainer import testing
+
+    if func_expected is None:
+        name = func.__name__
+        func_expected = getattr(numpy, name)
+
+    if make_data is None:
+        make_data = make_data_default
+
+    def f(klass):
+        assert issubclass(klass, unittest.TestCase)
+
+        def setUp(self):
+            self.x, self.gy = make_data(self.dtype, self.shape)
+            if self.dtype == numpy.float16:
+                self.backward_options = {
+                    'eps': 2 ** -4, 'atol': 2 ** -4, 'rtol': 2 ** -4,
+                    'dtype': numpy.float64}
+            else:
+                self.backward_options = {}
+        setattr(klass, "setUp", setUp)
+
+        def check_forward(self, x_data):
+            x = chainer.Variable(x_data)
+            y = func(x)
+            self.assertEqual(y.data.dtype, x_data.dtype)
+            y_expected = func_expected(cuda.to_cpu(x_data), dtype=x_data.dtype)
+            testing.assert_allclose(y_expected, y.data, atol=1e-4, rtol=1e-4)
+        setattr(klass, "check_forward", check_forward)
+
+        @condition.retry(3)
+        def test_forward_cpu(self):
+            self.check_forward(self.x)
+        setattr(klass, "test_forward_cpu", test_forward_cpu)
+
+        @attr.gpu
+        @condition.retry(3)
+        def test_forward_gpu(self):
+            self.check_forward(cuda.to_gpu(self.x))
+        setattr(klass, "test_forward_gpu", test_forward_gpu)
+
+        def check_backward(self, x_data, y_grad):
+            gradient_check.check_backward(
+                func, x_data, y_grad, **self.backward_options)
+        setattr(klass, "check_backward", check_backward)
+
+        @condition.retry(3)
+        def test_backward_cpu(self):
+            self.check_backward(self.x, self.gy)
+        setattr(klass, "test_backward_cpu", test_backward_cpu)
+
+        @attr.gpu
+        @condition.retry(3)
+        def test_backward_gpu(self):
+            self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
+        setattr(klass, "test_backward_gpu", test_backward_gpu)
+
+        def test_label(self):
+            klass = func_class(func)
+            self.assertEqual(klass().label, func.__name__)
```

Comment:
We can overwrite label name in subclass of chainer Functions. So, we should some way for developers to feed expected label name.
\